modules/programs/bat.nix

```diff
@@ -136,6 +136,7 @@ in {
     })
     {
       home.packages = [ cfg.package ] ++ cfg.extraPackages;
+      home.sessionVariables = { MANPAGER = lib.mkDefault "bat -p -lman"; };
```

Comment:
This isn't what was requested. This is still setting the variable for everyone automatically with no way to disable it. `mkDefault` just sets the option's value at a lower `mkOrder` so that higher priorities can override it, but they would have to be provided through the module system in the same eval.

Reply:
This PR intends to make things simpler for a majority of users out-of-the-box. Adding an option imposes a cost which does no longer fulfills that intent.

Reply:
> > with no way to disable it
> 
> That is not correct. People can set it to the default, despite them having previously stated a certain general preference for `bat` (via `bat.enable = true;`).
> 
> ```nix
> { MANPAGER = "less"; }
> ```

Right now `MANPAGER` is unset for me. I have `programs.bat` enabled. I do not want this change to set it for me. 

> This PR intends to make things simpler for a majority of users out-of-the-box. Adding an option imposes a cost which does no longer suficently fulfill that intent.

A lot of people just want a replacement for cat, too. Still should be an option to set an environment variable for an optional behavior. It only makes sense to forcibly enable things that are required for basic functionality.